normits_demand/models/forecasting/EDGE_growth/run.py

```diff
@@ -0,0 +1,315 @@
+            # convert back to zonal level demand
+            zonal_grown_demand_mx = (
+                utils.convert_stns_to_zonal_demand(
+                    np_stn2stn_grown_demand_mx,
+                    zonal_from_to_stns,
+                    time_period,
+                    to_home,
+                )
+            )
+            # add to grown matrices dictionary
+            factored_matrices[segment] = zonal_grown_demand_mx
+
+    return loading.TPLoopOutput(overall_not_grown_demand, overall_base_demand, factored_matrices, growth_summary)
+
+
+def run_edge_growth(params: forecast_cnfg.EDGEParameters) -> None:
+    """Run Growth Process."""
+    LOG.info("#" * 80)
+    LOG.info(
+        "Started Process @ %s",
+        timing.get_datetime("%d-%m-%Y  %H:%M:%S"),
+    )
+    LOG.info("#" * 80)
+    LOG.info("Loading global variables for growth process.")
+    global_params = loading.load_globals(params)
+    # loop over forecast years
+    for forecast_year in params.forecast_years:
+        LOG.info(
+            "**** Applying growth for %s @ %s",
+            forecast_year,
+            timing.get_datetime(),
+        )
+        # read input files
+        growth_factors = file_ops.read_df(
+            params.edge_growth_dir
+            / params.forecast_years[forecast_year]
+        )
+        # produce growth matrices
+        growth_matrix_dic = growth_matrices.prepare_growth_matrices(
+            global_params.demand_segments,
+            growth_factors,
+            global_params.station_tlcs,
+        )
+        # fill growth matrices
+        filled_growth_matrices= growth_matrices.fill_missing_factors(global_params.purposes,growth_matrix_dic)
+        # create empty dictionary to store matrices
+        factored_24hr_matrices = {}
+        # declare global demand total variables
+        # empty dataframe for growth summary
+
+        # loop over time periods
+        for tp in global_params.time_periods:
+            hdf_file = params.irsj_props_dir / f"{tp}_iRSj_probabilities_split.h5"
+            if not os.path.isfile(hdf_file):
+                utils.split_irsj(params.irsj_props_dir, 'userclass', tp)
+        tp_looper = partial(
+            _tp_loop,
+            params.matrices_to_grow_dir,
+            params.irsj_props_dir,
+            len(global_params.station_tlcs),
+            forecast_year,
+            global_params.demand_segments,
+            filled_growth_matrices,
+        )
+        args = [
+            (time_period, global_params.ticket_type_splits[time_period])
+            for time_period in global_params.time_periods
+        ]
+        outputs = multiprocess(
+            tp_looper, args, in_order=True
+        )
+        factored = [k.factored for k in outputs]
+        factored_matrices = dict(zip(global_params.time_periods, factored))
+        overall_not_grown_demand = sum([i.overall_not_grown for i in outputs])
+        overall_base_demand = sum([i.overall_base for i in outputs])
```

Comment:
Is this chunk of code just for converting matrices, growing, then converting back?
Can this be placed into a function to make this more obvious